clustered/server/ehcache-service/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainMap.java

```diff
@@ -354,12 +354,24 @@ public Lock writeLock() {
   protected void storageEngineFailure(Object failure) {
   }
 
-  static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
+  public static class HeadMap<K> extends EvictionListeningReadWriteLockedOffHeapClockCache<K, InternalChain> {
 
     public HeadMap(EvictionListener<K, InternalChain> listener, PageSource source, ChainStorageEngine<K> chainStorage) {
       super(listener, source, chainStorage);
     }
 
+    public void removeAtSlot(int slot, boolean shrink) {
```

Comment:
No _close_ unit test covers this.  More specifically, there is no unit test in this package that covers `OffHeapChainStorageEngine.AttachedInternalChain.removeHeader` from which this (`removeAtSlot`) method is called.  You might consider adding additional coverage for testing of `AttachedInternalChain` so this method actually gets hit.

Reply:
Will add an appropriate test.